ui/app/src/modules/Preview/Tools/ContentTree.tsx

```diff
@@ -0,0 +1,400 @@
+  const classes = treeItemStyles({});
+  const [over, setOver] = useState(false);
+  let timeout = React.useRef<any>();
+
+  let Icon;
+  if (nodes.type === 'page') {
+    Icon = Page;
+  } else if (nodes.type === 'node-selector') {
+    Icon = NodeSelector;
+  } else if (nodes.type === 'component') {
+    Icon = Component;
+  } else if (nodes.type === 'repeat') {
+    Icon = Repeat;
+  } else {
+    Icon = Field;
+  }
+
+  function setOverState(e: React.MouseEvent, isOver: boolean) {
+    e.stopPropagation();
+    clearTimeout(timeout.current);
+    if (!isOver) {
+      timeout.current = setTimeout(() => {
+        setOver(false);
+      }, 10)
+    } else {
+      setOver(isOver);
+    }
+  }
+
+  return (
+    <TreeItem
+      key={nodes.id}
+      nodeId={nodes.id}
+      onMouseOver={(e) => setOverState(e, true)}
+      onMouseOut={(e) => setOverState(e, false)}
+      label={
+        <div className={classes.treeItemLabel} onClick={() => handleScroll(nodes)}>
+          {
+            (nodes.id === 'root' && handlePrevious) ? (
+              <ChevronLeftRounded onClick={(e) => handlePrevious(e)}/>
+            ) : (
+              <>
+                {(nodes.type === 'component') && <ChevronRightIcon onClick={() => handleClick(nodes)}/>}
+                <Icon className={classes.icon}/>
+              </>
+            )
+          }
+          <p>{nodes.name}</p>
+          {
+            over &&
+            <IconButton className={classes.options} onMouseOver={(e) => setOverState(e, true)}
+                        onClick={(e) => handleOptions(e)}>
+              <MoreVertIcon/>
+            </IconButton>
+          }
+        </div>
+      }
+      classes={{
+        root: classes.treeItemRoot,
+        content: clsx(classes.treeItemContent, (!nodes.children?.length && nodes.type !== 'component') && 'padded'),
+        expanded: classes.treeItemExpanded,
+        group: classes.treeItemGroup,
+        iconContainer: nodes.id === 'root' ? classes.treeItemIconContainer : ''
+      }}
+    >
+      {
+        Array.isArray(nodes.children)
+          ? nodes.children.map(node =>
+            <TreeItemCustom
+              key={node.id}
+              nodes={node}
+              {...reversePluckProps(props, 'nodes')}
+            />
+          )
+          : null
+      }
+    </TreeItem>
+  )
+}
+
+
+export default function ContentTree() {
+  const classes = useStyles({});
+  const guest = usePreviewGuest();
+  const { formatMessage } = useIntl();
+  const contentTypesBranch = useSelection(state => state.contentTypes);
+  const hostToGuest$ = getHostToGuestBus();
+  const [expanded, setExpanded] = React.useState<string[]>(['root']);
+  const [data, setData] = React.useState<Data>({
+    previous: [],
+    selected: null,
+    lookupTable: null
+  });
+
+  useEffect(() => {
+    if (guest?.modelId && guest?.models && contentTypesBranch.byId && data.selected === null) {
```

Comment:
once you've done `guest?.`, on the subsequent uses is not necessary to use the opt chaining op as the first if would have checked for _nullish-ness_